### ISSUE TITLE
Deep Residual Learning for Image Recognition

### ISSUE THREAD

**Issue by guanfuchen** [guanfuchen/DeepNetModel#4]
related paper

|摘要|
|---|
|Deeper neural networks are more difficult to train. We present a residual learning framework to ease the training of networks that are substantially deeper than those used previously. We explicitly reformulate the layers as learning residual functions with reference to the layer inputs, instead of learning unreferenced functions. We provide comprehensive empirical evidence showing that these residual networks are easier to optimize, and can gain accuracy from considerably increased depth. On the ImageNet dataset we evaluate residual nets with a depth of up to 152 layers—8* deeper than VGG nets [41] but still having lower complexity. An ensemble of these residual nets achieves 3.57% error on the ImageNet test set. This result won the 1st place on the ILSVRC 2015 classification task. We also present analysis on CIFAR-10 with 100 and 1000 layers.
The depth of representations is of central importance for many visual recognition tasks. Solely due to our extremely deep representations, we obtain a 28% relative improvement on the COCO object detection dataset. Deep residual nets are foundations of our submissions to ILSVRC & COCO 2015 competitions1, where we also won the 1st places on the tasks of ImageNet detection, ImageNet localization, COCO detection, and COCO segmentation.|

**Comment by guanfuchen** [comment]
variants and experiment

![image](https://user-images.githubusercontent.com/22321977/48926974-68ffad80-ef0d-11e8-9ab4-c2d8bc245d96.png)

result

![image](https://user-images.githubusercontent.com/22321977/48926978-7157e880-ef0d-11e8-9e19-b8172dad015b.png)

![image](https://user-images.githubusercontent.com/22321977/48926997-8a609980-ef0d-11e8-87f8-c80f7fe98263.png)

![image](https://user-images.githubusercontent.com/22321977/48927000-8f254d80-ef0d-11e8-94ca-3e6fffc40477.png)

![image](https://user-images.githubusercontent.com/22321977/48927002-94829800-ef0d-11e8-828f-6d87bae2ec45.png)

![image](https://user-images.githubusercontent.com/22321977/48927005-98aeb580-ef0d-11e8-9fbe-920bfe572aa5.png)

![image](https://user-images.githubusercontent.com/22321977/48927011-9ea49680-ef0d-11e8-98f4-ebc233a6ddfb.png)
